engine/build.gradle

```diff
@@ -244,3 +244,8 @@ idea {
         downloadSources = true
     }
 }
+
+// Make sure our config file for code analytics get extracted (vulnerability: non-IDE execution of single analytic)
+ideaModule.dependsOn rootProject.extractConfig
+tasks.eclipse.dependsOn rootProject.extractConfig
+check.dependsOn rootProject.extractConfig
```

Comment:
Yep, it absolutely could fail, that's what I dislike about it :D

I wonder if me adding an if caused the slowdown. I had checkstyleMain etc trigger extractConfig in common.gradle, but *only* if it wasn't done in the root project - that way module builds wouldn't do it (they'd get the config files via engine harness)

If they tried to call exactConfig it would fail since they wouldn't have that task (they'd start with Core's build.gradle). Although ... maybe I could put the task itself into common.gradle.

I figured even with the if the task would only get added to the execution once, but it slowed each module tasks config step by about 5 seconds. At that point the zipping shouldn't be happening yet so I wouldn't think that would be the slowdown.

In any case I'll try tweaking at it more later, and yeah at least using [analytic]Main instead of check would capture the three analytics independently rather than just the global version (but still not at the per-module level)